Ambiguous beans in TCK

With the changes in #742, I'm now seeing an issue running the TCK:

All the tests in ConverterTest fail due to this error when arquillian attempts to inject the test class:
```
Caused by: org.jboss.arquillian.test.spi.ArquillianProxyException: org.jboss.weld.exceptions.DeploymentException : WELD-001409: Ambiguous dependencies for type InjectingBean with qualifiers @Default
  at injection point [BackedAnnotatedField] @Inject org.eclipse.microprofile.config.tck.ConverterTest.bean
  at org.eclipse.microprofile.config.tck.ConverterTest.bean(ConverterTest.java:0)
  Possible dependencies: 
  - Managed Bean [class org.eclipse.microprofile.config.tck.ConverterTest$InjectingBean] with qualifiers [@Any @Default],
  - Managed Bean [class org.eclipse.microprofile.config.tck.ConverterTest$InjectingBean] with qualifiers [@Any @Default]
```

I suspect this is because the `ConverterTest` class is added to the application twice (once in the deployment method and once automatically because it is the test class).

We use the same pattern elsewhere in `CDIPlainInjectionTest`, `CDIPropertyExpressionsTest` and `ConfigValueTest` and it took me a while to work out why I wasn't seeing any issues in those tests.

The difference in those tests is that the deployment method creates a single .war archive. In `ConverterTest` we create a .jar inside a .war. The .jar has a beans.xml and the .war does not. This means you'll only see this issue on a CDI Full implementation where an archive without a beans.xml is scanned for bean annotations.

Testing locally, I can fix this issue either by changing `ConverterTest` to create a single archive, or by moving the `InjectionBean` into its own file.

--- a/tck/src/main/java/org/eclipse/microprofile/config/tck/ConverterTest.java
+++ b/tck/src/main/java/org/eclipse/microprofile/config/tck/ConverterTest.java
@@ -57,7 +57,6 @@ import org.jboss.arquillian.container.test.api.Deployment;
 import org.jboss.arquillian.testng.Arquillian;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.asset.EmptyAsset;
-import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
 import org.testng.Assert;
 import org.testng.annotations.Test;
@@ -77,8 +76,8 @@ public class ConverterTest extends Arquillian {
 
     @Deployment
     public static WebArchive deploy() {
-        JavaArchive testJar = ShrinkWrap
-                .create(JavaArchive.class, "converterTest.jar")
+        WebArchive testWar = ShrinkWrap
+                .create(WebArchive.class, "converterTest.war")
                 .addClass(ConverterTest.class)
                 .addPackage(CustomDbConfigSource.class.getPackage())
                 .addClasses(DuckConverter.class, Duck.class, Donald.class, UpperCaseDuckConverter.class,
@@ -87,19 +86,16 @@ public class ConverterTest extends Arquillian {
                 .addAsServiceProvider(ConfigSource.class, CustomDbConfigSource.class)
                 .addAsServiceProvider(ConfigSourceProvider.class, CustomConfigSourceProvider.class)
                 .addAsServiceProvider(Converter.class, DuckConverter.class)
-                .as(JavaArchive.class);
+                .as(WebArchive.class);
 
-        AbstractTest.addFile(testJar, "META-INF/microprofile-config.properties");
-        AbstractTest.addFile(testJar, "sampleconfig.yaml");
+        AbstractTest.addFile(testWar, "META-INF/microprofile-config.properties");
+        AbstractTest.addFile(testWar, "sampleconfig.yaml");
 
-        WebArchive war = ShrinkWrap
-                .create(WebArchive.class, "converterTest.war")
-                .addAsLibrary(testJar);
-        return war;
+        return testWar;
     }
 
     @Inject
-    InjectingBean bean;
+    private InjectingBean bean;
 
     @Test(expectedExceptions = IllegalArgumentException.class)
     public void testDonaldNotConvertedByDefault() {

